src/libstd/sys/unix/condvar.rs

```diff
@@ -92,7 +92,16 @@ impl Condvar {
         assert_eq!(r, 0);
 
         // Nanosecond calculations can't overflow because both values are below 1e9.
-        let nsec = dur.subsec_nanos() as libc::c_long + now.tv_nsec as libc::c_long;
+        // HACK: add support to linux x32, the original code was
```

Comment:
Is this necessary b/c the values are always < 1e9? Doesn't that mean we can basically cast to "anything reasonable" here and expect it to work?

Reply:
Should we use i32?

Reply:
I think that could work? It just seemed like the extra casting wasn't necessary here given the various constraints.

Reply:
As subsec_nanos is u32, I opted to use u32. Both i32 and u32 are fine.